plugins/modules/ndo_mac_sec_policy.py

```diff
@@ -0,0 +1,443 @@
+    description: "Ansible Test MACSec Policy"
+    mac_sec_key:
+      - key_name: ansible_test_key
+        psk: 'AA111111111111111111111111111111111111111111111111111111111111aa'
+        start_time: '2029-12-11 11:12:13'
+        end_time: 'infinite'
+    state: present
+
+- name: Query a MACSec Policy with mac_sec_policy name
+  cisco.mso.ndo_mac_sec_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_test_template
+    mac_sec_policy: ansible_test_mac_sec_policy
+    state: query
+  register: query_one
+
+- name: Query all MACSec Policies
+  cisco.mso.ndo_mac_sec_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_test_template
+    state: query
+  register: query_all
+
+- name: Query a MACSec Policy with mac_sec_policy UUID
+  cisco.mso.ndo_mac_sec_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_test_template
+    mac_sec_policy_uuid: ansible_test_mac_sec_policy_uuid
+    state: query
+  register: query_uuid
+
+- name: Delete a MACSec Policy with name
+  cisco.mso.ndo_mac_sec_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_test_template
+    mac_sec_policy: ansible_test_mac_sec_policy
+    state: absent
+
+- name: Delete a MACSec Policy with UUID
+  cisco.mso.ndo_mac_sec_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_test_template
+    mac_sec_policy_uuid: ansible_test_mac_sec_policy_uuid
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec
+from ansible_collections.cisco.mso.plugins.module_utils.template import MSOTemplate, KVPair
+from ansible_collections.cisco.mso.plugins.module_utils.constants import NDO_CIPHER_SUITE_MAP, NDO_SECURITY_POLICY_MAP
+import copy
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        dict(
+            template=dict(type="str", required=True),
+            mac_sec_policy=dict(type="str", aliases=["name"]),
+            mac_sec_policy_uuid=dict(type="str", aliases=["uuid"]),
+            description=dict(type="str"),
+            admin_state=dict(type="str", choices=["enabled", "disabled"]),
+            interface_type=dict(type="str", choices=["fabric", "access"], default="fabric"),
+            cipher_suite=dict(type="str", choices=["128_gcm_aes", "128_gcm_aes_xpn", "256_gcm_aes", "256_gcm_aes_xpn"]),
+            window_size=dict(type="int"),
+            security_policy=dict(type="str", choices=["should_secure", "must_secure"]),
+            sak_expiry_time=dict(type="int"),
+            confidentiality_offset=dict(type="int", choices=[0, 30, 50]),
+            key_server_priority=dict(type="int"),
+            mac_sec_key=dict(
```

Comment:
How do you update the existing mac_sec_key?  When we have multiple keys in the list, but if I want to update a specific one in the list.

@lhercot Can we have a separate module for the `mac_sec_key` because it is a list of items?

Reply:
for now we will keep the mac_sec_keys in this module.